### ISSUE TITLE
[UE 5.1] The following modules are missing or built with a different engine version

### ISSUE THREAD

**Issue by weepingforest** [ValveSoftware/steam-audio#341]
I'm trying to get Steam Audio to work with Unreal Engine 5. The problem is that whenever I go to Plugins section in Unreal Engine 5 and enable Steam Audio, it requires me to restart UE5, and when I do it always gives me this error:
"The following modules are missing or built with a different engine version:
Steam Audio
SteamAudioEditor
Engine modules cannot be compiled at runtime. Please build through your IDE."

And the only option available is 'OK' which does nothing.

First time I saw this error was in Unreal Engine 5.3.2, so I guessed it was because of lack of compatibility. So now I tried it with UE 5.1.0, but I still get the same error. I downloaded the plugin directly and didn't build it myself with IDE. Needless to say, I also deleted the default Steam Audio plugin from Plugins/Runtime/Steam folder.
I tried it with Steam Audio 4.5.3 and 4.1.3. I get the same error with both.
Thanks in advance.

**Comment by lakulish** [comment]
@weepingforest Are you installing the `SteamAudio` plugin into the `Plugins` subdirectory of your project, or into the `Engine/Plugins` subdirectory of your Unreal Engine installation directory? UE5 should be able to recompile the plugin if it's installed into the project-specific plugins subdirectory.

**Comment by FraBad98** [comment]
@weepingforest I don't have the solution, unfortunately. I'm here because I'm having the exact same issue. Did you manage to fix it? If so, what was the procedure?

Thanks in advance!

**Comment by LinusLindblad** [comment]
@FraBad98 In-case you or anyone else that stumbles on this thread has the same issue, this is caused by placing the steamaudio plugin into the plugin folder of your **engine directory** [UE5 Root]/Engine/Plugins/[Plugin Name]. Instead of the per-project plugin folder located at [Project Root]/Plugins/. 

If there is no "plugins" folder inside your project root you can add one.

Unreal will report the same error on initial startup, but will actually be able to compile the modules now.

(If this doesn't work for you or there are issues stopping the project from opening, you can edit the .uproject file in notepad to stop the plugin from loading externally.)

Hope this helps!

**Comment by wolfsprite** [comment]
> @FraBad98 In-case you or anyone else that stumbles on this thread has the same issue, this is caused by placing the steamaudio plugin into the plugin folder of your **engine directory** [UE5 Root]/Engine/Plugins/[Plugin Name]. Instead of the per-project plugin folder located at [Project Root]/Plugins/.

I actually just tried this, including deleting the Engine's SteamAudio plugin (as the readme included with the SDK recommends), but I'm also running into the issue. :(

**Comment by wolfsprite** [comment]
Made some progress by adding the folder to a C++ project's Plugins folder while the Engine was open and manually telling it to recompile, though I also had to change a nested if statement in SteamAudioSourceComponent.cpp due to the compiler erroring out on its indentation:
![image](https://github.com/user-attachments/assets/ca231a2c-ea12-48af-b208-612fd2556ac6) 

But I'm now running into an error about a missing library I can't find, and I'm not yet sure how to resolve:

![image](https://github.com/user-attachments/assets/20f8777f-df26-490b-880c-43f89fa7c4fb)

`ld.lld: error: unable to find library -lUnrealEditor-SteamAudio`


[using UE 5.4.2]

**Comment by noidedxyz** [comment]
Adding brackets for some reason fixed this issue for me. I don't know why it's throwing a fuss about this, tbh. 

```
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, OcclusionType)))
    {
        return bParentVal && bSimulateOcclusion;
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, OcclusionRadius)))
    {
        return bParentVal && bSimulateOcclusion && (OcclusionType == EOcclusionType::VOLUMETRIC);
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, OcclusionSamples)))
    {
        return bParentVal && bSimulateOcclusion && (OcclusionType == EOcclusionType::VOLUMETRIC);
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, OcclusionValue)))
    {
        return bParentVal && !bSimulateOcclusion;
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, bSimulateTransmission)))
    {
        return bParentVal && bSimulateOcclusion;
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, TransmissionLowValue)))
    {
        return bParentVal && !bSimulateTransmission;
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, TransmissionMidValue)))
    {
        return bParentVal && !bSimulateTransmission;
    }
    if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, TransmissionHighValue)))
    {
        return bParentVal && !bSimulateTransmission;
    }
	if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, ReflectionsType)))
    {
		return bParentVal && bSimulateReflections;
    }
	if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, CurrentBakedSource)))
    {
		return bParentVal && bSimulateReflections && (ReflectionsType == EReflectionSimulationType::BAKED_STATIC_SOURCE);
    }
	if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, PathingProbeBatch)))
    {
		return bParentVal && bSimulatePathing;
    }
	if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, bPathValidation)))
    {
		return bParentVal && bSimulatePathing;
    }
	if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, bFindAlternatePaths)))
    {
		return bParentVal && bSimulatePathing;
    }
```

**Comment by wolfsprite** [comment]
> Adding brackets for some reason fixed this issue for me. I don't know why it's throwing a fuss about this, tbh. 

Yeah.. I had to do that too.. Which fixed one issue, but I'm still running into this unfortunately:
`ld.lld: error: unable to find library -lUnrealEditor-SteamAudio`

**Comment by lakulish** [comment]
@wolfsprite Can you delete the `Plugins/SteamAudio/Binaries` and `Plugins/SteamAudio/Intermediate` directories in your Unreal project, and try building again? If you still encounter errors, could you provide the complete build log? Thanks!

**Comment by wolfsprite** [comment]
> @wolfsprite Can you delete the `Plugins/SteamAudio/Binaries` and `Plugins/SteamAudio/Intermediate` directories in your Unreal project, and try building again? If you still encounter errors, could you provide the complete build log? Thanks!

Tried a fresh project on 5.4.4, deleted the directories you mentioned, and attempted build again in-Engine (after fixing USteamAudioSourceComponent::CanEditChange in SteamAudioSourceComponent.cpp, as per above comments); but am getting the same error:
```sh
[22/23] Link (lld) libUnrealEditor-SteamAudioEditor-0001.so
ld.lld: error: unable to find library -lUnrealEditor-SteamAudio
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```
I also attempted to manually rerun the UnrealBuildTool command UE used, but with the engine closed, and with -v and -verbose flags to try to get more details, but I'm not sure where in the process I'd be able to get clang the "use -v to see invocation" parameter it needs. (I'm actually rather new to C++..) It resulted in the same error, either way. 
Here is the full build log (with personal info stripped):
```sh
Setting up bundled DotNet SDK
Running command : dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll -ModuleWithSuffix=Test54,5337 Test54Editor Linux Development -Project=/home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Test54.uproject -IgnoreJunk -progress -verbose
Log file: /home/username/.config/Epic/UnrealBuildTool/Log.txt
Creating makefile for Test54Editor (no existing makefile)
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 33%
Parsing headers for Test54Editor
  Running Internal UnrealHeaderTool /home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Intermediate/Build/Linux/Test54Editor/Development/Test54Editor.uhtmanifest -WarningsAsErrors -installed
Total of 62 written
Reflection code generated for Test54Editor in 1.4160261 seconds
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
------- Build details --------
Using toolchain located at '/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/clang++) version 'clang version 16.0.6 (github.com/llvm/llvm-project 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin' (string), 16 (major), 0 (minor), 6 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar) version 'LLVM (http://llvm.org/):
  LLVM version 16.0.6
  Optimized build. (string)'
Using fast way to relink  circularly dependent libraries (no FixDeps).
Targeted minimum CPU architecture: None
------------------------------
Building Test54Editor...
[Adaptive Build] Excluded from SteamAudio unity file: SteamAudioGeometryComponent.cpp, SteamAudioSourceComponent.cpp
Determining max actions to execute in parallel (8 physical cores, 16 logical cores)
  Executing up to 8 processes, one per physical core
Using Parallel executor to run 23 action(s)
@progress 'Compiling C++ source code...' 0%
------ Building 23 action(s) started ------
@progress 'Compiling C++ source code...' 4%
@progress 'Compiling C++ source code...' 9%
[1/23] Compile SharedPCH.Engine.Project.ValApi.Cpp20.h
@progress 'Compiling C++ source code...' 13%
[2/23] Compile Test54.cpp
@progress 'Compiling C++ source code...' 17%
[3/23] Compile SteamAudioGeometryComponent.cpp
@progress 'Compiling C++ source code...' 22%
[4/23] Compile TP_WeaponComponent.gen.cpp
@progress 'Compiling C++ source code...' 26%
[5/23] Compile SteamAudioSourceComponent.cpp
@progress 'Compiling C++ source code...' 30%
[6/23] Compile TP_PickUpComponent.cpp
@progress 'Compiling C++ source code...' 35%
[7/23] Compile TP_PickUpComponent.gen.cpp
@progress 'Compiling C++ source code...' 39%
[8/23] Compile Test54.init.gen.cpp
@progress 'Compiling C++ source code...' 43%
[9/23] Compile TP_WeaponComponent.cpp
@progress 'Compiling C++ source code...' 48%
[10/23] Compile Test54GameMode.gen.cpp
@progress 'Compiling C++ source code...' 52%
[11/23] Compile Test54Character.gen.cpp
@progress 'Compiling C++ source code...' 57%
[12/23] Compile Test54GameMode.cpp
@progress 'Compiling C++ source code...' 61%
[13/23] Compile Test54PlayerController.gen.cpp
@progress 'Compiling C++ source code...' 65%
[14/23] Compile Test54PlayerController.cpp
@progress 'Compiling C++ source code...' 70%
[15/23] Compile Test54Character.cpp
@progress 'Compiling C++ source code...' 74%
[16/23] Compile Test54Projectile.cpp
@progress 'Compiling C++ source code...' 78%
[17/23] Compile Test54Projectile.gen.cpp
[18/23] Link (lld) libUnrealEditor-Test54-5337.so
@progress 'Compiling C++ source code...' 83%
@progress 'Compiling C++ source code...' 87%
[19/23] Compile Module.SteamAudio.cpp
[20/23] Link (lld) libUnrealEditor-SteamAudio-0001.so
@progress 'Compiling C++ source code...' 91%
@progress 'Compiling C++ source code...' 96%
[21/23] Compile Module.SteamAudioEditor.cpp
[22/23] Link (lld) libUnrealEditor-SteamAudioEditor-0001.so
ld.lld: error: unable to find library -lUnrealEditor-SteamAudio
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
Total time in Parallel executor: 47.75 seconds
Total execution time: 53.05 seconds
```

I'm also providing an additional build log from an attempt made after that, where I replaced the Editor target with the project's main target. This build *was* successful, but unfortunately did not allow the Engine/Editor to run successfully while the SteamAudio plugin was enabled:

```sh
Setting up bundled DotNet SDK
Running command : dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll -ModuleWithSuffix=Test54,5337 Test54 Linux Development -Project=/home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Test54.uproject -IgnoreJunk -progress -verbose
Log file: /home/username/.config/Epic/UnrealBuildTool/Log.txt
Creating makefile for Test54 (no existing makefile)
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 33%
Parsing headers for Test54
  Running Internal UnrealHeaderTool /home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Intermediate/Build/Linux/Test54/Development/Test54.uhtmanifest -WarningsAsErrors -installed
Total of 50 written
Reflection code generated for Test54 in 1.4877399 seconds
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
------- Build details --------
Using toolchain located at '/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/clang++) version 'clang version 16.0.6 (github.com/llvm/llvm-project 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin' (string), 16 (major), 0 (minor), 6 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar) version 'LLVM (http://llvm.org/):
  LLVM version 16.0.6
  Optimized build. (string)'
Using fast way to relink  circularly dependent libraries (no FixDeps).
Targeted minimum CPU architecture: None
------------------------------
Building Test54...
[Adaptive Build] Excluded from SteamAudio unity file: SteamAudioGeometryComponent.cpp, SteamAudioSourceComponent.cpp
Determining max actions to execute in parallel (8 physical cores, 16 logical cores)
  Executing up to 8 processes, one per physical core
Using Parallel executor to run 20 action(s)
@progress 'Compiling C++ source code...' 0%
------ Building 20 action(s) started ------
@progress 'Compiling C++ source code...' 5%
@progress 'Compiling C++ source code...' 10%
[1/20] Compile SharedPCH.Engine.Project.ValApi.Cpp20.h
@progress 'Compiling C++ source code...' 15%
[2/20] Compile Test54.init.gen.cpp
@progress 'Compiling C++ source code...' 20%
[3/20] Compile TP_WeaponComponent.gen.cpp
@progress 'Compiling C++ source code...' 25%
[4/20] Compile Test54Character.gen.cpp
@progress 'Compiling C++ source code...' 30%
[5/20] Compile Test54.cpp
@progress 'Compiling C++ source code...' 35%
[6/20] Compile TP_PickUpComponent.cpp
@progress 'Compiling C++ source code...' 40%
[7/20] Compile TP_PickUpComponent.gen.cpp
@progress 'Compiling C++ source code...' 45%
[8/20] Compile Test54Character.cpp
@progress 'Compiling C++ source code...' 50%
[9/20] Compile TP_WeaponComponent.cpp
@progress 'Compiling C++ source code...' 55%
[10/20] Compile Test54GameMode.cpp
@progress 'Compiling C++ source code...' 60%
[11/20] Compile Test54GameMode.gen.cpp
@progress 'Compiling C++ source code...' 65%
[12/20] Compile Test54PlayerController.gen.cpp
@progress 'Compiling C++ source code...' 70%
[13/20] Compile Test54Projectile.gen.cpp
@progress 'Compiling C++ source code...' 75%
[14/20] Compile Test54Projectile.cpp
@progress 'Compiling C++ source code...' 80%
[15/20] Compile Test54PlayerController.cpp
@progress 'Compiling C++ source code...' 85%
[16/20] Compile SteamAudioGeometryComponent.cpp
@progress 'Compiling C++ source code...' 90%
[17/20] Compile SteamAudioSourceComponent.cpp
@progress 'Compiling C++ source code...' 95%
[18/20] Compile Module.SteamAudio.cpp
[19/20] Link (lld) Test54
@progress 'Compiling C++ source code...' 100%
[20/20] WriteMetadata Test54.target
Total time in Parallel executor: 41.76 seconds
Total execution time: 46.34 seconds

```

**Comment by wolfsprite** [comment]
As I was not satisfied leaving it there, I want to submit one more build log, from an attempt I made just now that _seemed_ like it was going to work. The difference was adding `-Module=SteamAudio` and `-Module=<ProjectName>` to the build command:

```sh
$ steam-run /home/username/UnrealEngine/5.4.4/Engine/Build/BatchFiles/Linux/Build.sh -ModuleWithSuffix=Test54,5337 Test54 Test54Editor Linux Development -Module=SteamAudio -Module=Test54 -Project="/home/username/UnrealProjectsSelf/Test54/Test54.uproject" "/home/username/UnrealProjectsSelf/Test54/Test54.uproject" -IgnoreJunk -progress -verbose
Setting up bundled DotNet SDK
Running command : dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll -ModuleWithSuffix=Test54,5337 Test54 Test54Editor Linux Development -Module=SteamAudio -Module=Test54 -Project=/home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Test54.uproject -IgnoreJunk -progress -verbose
Log file: /home/username/.config/Epic/UnrealBuildTool/Log.txt
Creating makefile for Test54 (no existing makefile)
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 33%
Parsing headers for Test54
  Running Internal UnrealHeaderTool /home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Intermediate/Build/Linux/Test54/Development/Test54.uhtmanifest -WarningsAsErrors -installed
Total of 50 written
Reflection code generated for Test54 in 1.2993797 seconds
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
------- Build details --------
Using toolchain located at '/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/clang++) version 'clang version 16.0.6 (github.com/llvm/llvm-project 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin' (string), 16 (major), 0 (minor), 6 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar) version 'LLVM (http://llvm.org/):
  LLVM version 16.0.6
  Optimized build. (string)'
Using fast way to relink  circularly dependent libraries (no FixDeps).
Targeted minimum CPU architecture: None
------------------------------
Creating makefile for Test54Editor (no existing makefile)
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 33%
Parsing headers for Test54Editor
  Running Internal UnrealHeaderTool /home/username/UnrealProjectsSelf/Test54/Test54.uproject /home/username/UnrealProjectsSelf/Test54/Intermediate/Build/Linux/Test54Editor/Development/Test54Editor.uhtmanifest -WarningsAsErrors -installed
Total of 62 written
Reflection code generated for Test54Editor in 1.5707331 seconds
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
------- Build details --------
Using toolchain located at '/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/clang++) version 'clang version 16.0.6 (github.com/llvm/llvm-project 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin' (string), 16 (major), 0 (minor), 6 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar (/home/username/UnrealEngine/5.4.4/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v22_clang-16.0.6-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar) version 'LLVM (http://llvm.org/):
  LLVM version 16.0.6
  Optimized build. (string)'
Using fast way to relink  circularly dependent libraries (no FixDeps).
Targeted minimum CPU architecture: None
------------------------------
Building Test54 and Test54Editor...
Test54: [Adaptive Build] Excluded from SteamAudio unity file: SteamAudioGeometryComponent.cpp, SteamAudioSourceComponent.cpp
Test54Editor: [Adaptive Build] Excluded from SteamAudio unity file: SteamAudioGeometryComponent.cpp, SteamAudioSourceComponent.cpp
Determining max actions to execute in parallel (8 physical cores, 16 logical cores)
  Executing up to 8 processes, one per physical core
Using Parallel executor to run 23 action(s)
@progress 'Compiling C++ source code...' 0%
------ Building 23 action(s) started ------
@progress 'Compiling C++ source code...' 4%
** For Test54-Linux-Development **
@progress 'Compiling C++ source code...' 9%
[1/23] Compile SharedPCH.Engine.Project.ValApi.Cpp20.h
@progress 'Compiling C++ source code...' 13%
[2/23] Compile Test54.init.gen.cpp
@progress 'Compiling C++ source code...' 17%
[3/23] Compile TP_WeaponComponent.gen.cpp
@progress 'Compiling C++ source code...' 22%
[4/23] Compile Test54.cpp
@progress 'Compiling C++ source code...' 26%
[5/23] Compile TP_PickUpComponent.cpp
@progress 'Compiling C++ source code...' 30%
[6/23] Compile TP_PickUpComponent.gen.cpp
** For Test54Editor-Linux-Development **
@progress 'Compiling C++ source code...' 35%
[7/23] Compile SharedPCH.Engine.Project.ValApi.Cpp20.h
** For Test54-Linux-Development **
@progress 'Compiling C++ source code...' 39%
[8/23] Compile Test54Character.cpp
@progress 'Compiling C++ source code...' 43%
[9/23] Compile Test54Character.gen.cpp
@progress 'Compiling C++ source code...' 48%
[10/23] Compile TP_WeaponComponent.cpp
@progress 'Compiling C++ source code...' 52%
[11/23] Compile Test54GameMode.cpp
@progress 'Compiling C++ source code...' 57%
[12/23] Compile Test54GameMode.gen.cpp
@progress 'Compiling C++ source code...' 61%
[13/23] Compile Test54PlayerController.gen.cpp
@progress 'Compiling C++ source code...' 65%
[14/23] Compile Test54PlayerController.cpp
** For Test54Editor-Linux-Development **
@progress 'Compiling C++ source code...' 70%
[15/23] Compile SteamAudioGeometryComponent.cpp
** For Test54-Linux-Development **
@progress 'Compiling C++ source code...' 74%
[16/23] Compile Test54Projectile.gen.cpp
@progress 'Compiling C++ source code...' 78%
[17/23] Compile Test54Projectile.cpp
** For Test54Editor-Linux-Development **
@progress 'Compiling C++ source code...' 83%
[18/23] Compile SteamAudioSourceComponent.cpp
** For Test54-Linux-Development **
@progress 'Compiling C++ source code...' 87%
[19/23] Compile SteamAudioGeometryComponent.cpp
@progress 'Compiling C++ source code...' 91%
[20/23] Compile SteamAudioSourceComponent.cpp
@progress 'Compiling C++ source code...' 96%
[21/23] Compile Module.SteamAudio.cpp
** For Test54Editor-Linux-Development **
@progress 'Compiling C++ source code...' 100%
[22/23] Compile Module.SteamAudio.cpp
[23/23] Link (lld) libUnrealEditor-SteamAudio-0001.so
Total time in Parallel executor: 37.47 seconds
Total execution time: 44.53 seconds
```

**It appears to have been able to find and link the "UnrealEditor-SteamAudio" library that it was not able to before.** However, attempting to open the project resulted in bringing me back to the original issue in the OP:

```
LogLinux: Warning: MessageBox: The following modules are missing or built with a different engine version:

  Test54
  SteamAudio
  SteamAudioEditor
```

I'm not sure where the issue lies, but I'm hopeful that something in these logs is useful to whoever has the capacity to troubleshoot it! :)

**Comment by lakulish** [comment]
@wolfsprite The line:

```
[22/23] Link (lld) libUnrealEditor-SteamAudioEditor-0001.so
```

seems to indicate that you're building the plugin with the Unreal editor running, and the hot reload mechanism is getting confused. This is why there's a `-0001` at the end of the file name.

Can you try compiling with the editor fully closed? You should see a line that says `Link (lld) libUnrealEditor-SteamAudioEditor.so`. This is without adding any `-Module` parameters to the command.

**Comment by wolfsprite** [comment]
>This is why there's a `-0001` at the end of the file name.

That I didn't know; super helpful, thanks! 
I had thought the engine _was_ closed, and was not aware hotreload was interfering.

> Can you try compiling with the editor fully closed? You should see a line that says `Link (lld) libUnrealEditor-SteamAudioEditor.so`. This is without adding any `-Module` parameters to the command.

I have now done so and that appears to have worked perfectly! Compilation was successful and I can now open the project, and the Steam Audio plugin is enabled with its settings available in Project Settings. :D

Here's the log for that:
```sh
steam-run /home/username/UnrealEngine/5.4.4/Engine/Build/BatchFiles/Linux/Build.sh Test54Editor Linux Development -Project="/home/username/UnrealProjectsSelf/Test54/Test54.uproject" -IgnoreJunk -progress -verbose
Setting up bundled DotNet SDK
Running command : dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll Test54Editor Linux Development -Project=/home/username/UnrealProjectsSelf/Test54/Test54.uproject -IgnoreJunk -progress -verbose
Log file: /home/username/.config/Epic/UnrealBuildTool/Log.txt
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
Building Test54Editor...
[Adaptive Build] Excluded from SteamAudio unity file: SteamAudioGeometryComponent.cpp, SteamAudioSourceComponent.cpp
Determining max actions to execute in parallel (8 physical cores, 16 logical cores)
  Executing up to 8 processes, one per physical core
Using Parallel executor to run 23 action(s)
@progress 'Compiling C++ source code...' 0%
------ Building 23 action(s) started ------
@progress 'Compiling C++ source code...' 4%
@progress 'Compiling C++ source code...' 9%
[1/23] Compile SharedPCH.Engine.Project.ValApi.Cpp20.h
@progress 'Compiling C++ source code...' 13%
[2/23] Compile Test54.cpp
@progress 'Compiling C++ source code...' 17%
[3/23] Compile SteamAudioGeometryComponent.cpp
@progress 'Compiling C++ source code...' 22%
[4/23] Compile TP_WeaponComponent.gen.cpp
@progress 'Compiling C++ source code...' 26%
[5/23] Compile TP_PickUpComponent.cpp
@progress 'Compiling C++ source code...' 30%
[6/23] Compile SteamAudioSourceComponent.cpp
@progress 'Compiling C++ source code...' 35%
[7/23] Compile TP_PickUpComponent.gen.cpp
@progress 'Compiling C++ source code...' 39%
[8/23] Compile Test54.init.gen.cpp
@progress 'Compiling C++ source code...' 43%
[9/23] Compile TP_WeaponComponent.cpp
@progress 'Compiling C++ source code...' 48%
[10/23] Compile Test54GameMode.gen.cpp
@progress 'Compiling C++ source code...' 52%
[11/23] Compile Test54Character.gen.cpp
@progress 'Compiling C++ source code...' 57%
[12/23] Compile Test54GameMode.cpp
@progress 'Compiling C++ source code...' 61%
[13/23] Compile Test54PlayerController.gen.cpp
@progress 'Compiling C++ source code...' 65%
[14/23] Compile Test54PlayerController.cpp
@progress 'Compiling C++ source code...' 70%
[15/23] Compile Test54Character.cpp
@progress 'Compiling C++ source code...' 74%
[16/23] Compile Test54Projectile.cpp
@progress 'Compiling C++ source code...' 78%
[17/23] Compile Test54Projectile.gen.cpp
[18/23] Link (lld) libUnrealEditor-Test54.so
@progress 'Compiling C++ source code...' 83%
@progress 'Compiling C++ source code...' 87%
[19/23] Compile Module.SteamAudio.cpp
[20/23] Link (lld) libUnrealEditor-SteamAudio.so
@progress 'Compiling C++ source code...' 91%
@progress 'Compiling C++ source code...' 96%
[21/23] Compile Module.SteamAudioEditor.cpp
[22/23] Link (lld) libUnrealEditor-SteamAudioEditor.so
@progress 'Compiling C++ source code...' 100%
[23/23] WriteMetadata Test54Editor.target
Total time in Parallel executor: 47.42 seconds
Total execution time: 48.16 seconds
```

**Comment by WhitleyStriber** [comment]
[log.txt](https://github.com/user-attachments/files/18384429/log.txt)

Having issues compiling under Linux 5.5.1, also 5.5.0. 

 ```error: misleading indentation; statement is not part of the previous 'if' [-Werror,-Wmisleading-indentation]
  219 |         if ((InProperty->GetFName() == GET_MEMBER_NAME_CHECKED(USteamAudioSourceComponent, ReflectionsType)))```
